src/passwordcheck.js

```diff
@@ -28,29 +36,26 @@ function ($, bloomfilter) {
     return regex.test(password);
   }
 
-  return function checkPasswordStrength(password, callback) {
-    if (password.length < 8) {
-      return;
-    }
-    var bool = false;
-    var strong = strengthCheck(password.toString());
-
-    if (strong) {
-      // Only if the password has a chance of being strong do we check with the bloom filter
-      // else, simply reject the password. This helps us to not store all-alpha or all-numeric passwords
-      // on the bloom filter, reducing space.
-      bool = bloomfilter.test(password);
-      callback('Password strength check with bloomfilter triggered');
-    } else {
-      bool = true;
-      callback('Password with all alphabets/numbers detected');
-    }
+  return function (password, callback) {
+    if (password && typeof password === 'string' && password.length > 8) {
+      var isPasswordWeak = false;
```

Comment:
`isPasswordWeak` doesn't really serve a purpose way out here, nor on line 56 since nothing is done with the value after it is set. It's only really useful on lines 49 & 50. I'd move the `var` declaration to line 49.